### PR TITLE
[native] Add e2e tests for accessing fields of a struct type

### PR DESCRIPTION
**Pull request by pdabre12**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by aditi-pandit** [comment]
@pdabre12 : The intent was to add e2e tests for https://github.com/facebookincubator/velox/issues/6062 also. Please can you check if that works now.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 1]
```diff
@@ -1339,6 +1339,13 @@ public void testMergeEmptyHll()
         assertQuery("select cardinality(merge(empty_approx_set(0.1))) from orders");
     }
 
+    @Test
+    public void testStructureTypeData()
```
Comment (aditi-pandit) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:1343]: Nit : Rename function to testStructFieldAccess()

Comment (mbasmanova) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:1343]: @kevinwilfong Kevin already added testDereference in #20727. This is not needed.

Comment (aditi-pandit) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:1343]: @mbasmanova : We had found another problematic SQL in our tests `SELECT r[2] FROM (VALUES (ROW (ROW (1, 'a', true)))) AS v(r)`
@pdabre12 : The above SQL is what we want in the test. You can add it in the same test as Kevin in https://github.com/prestodb/presto/pull/20727

**Comment by pdabre12** [comment]
@aditi-pandit  Made a mistake , opening a new PR : #20814

**Pull request closed without merging**